pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/MethodNameDeclaration.java

```diff
@@ -10,18 +10,18 @@
 import net.sourceforge.pmd.internal.util.IteratorUtil;
 import net.sourceforge.pmd.lang.java.ast.ASTFormalParameter;
 import net.sourceforge.pmd.lang.java.ast.ASTFormalParameters;
-import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclarator;
+import net.sourceforge.pmd.lang.java.ast.ASTMethodDeclaration;
 import net.sourceforge.pmd.lang.java.ast.ASTType;
 import net.sourceforge.pmd.lang.symboltable.AbstractNameDeclaration;
 
 public class MethodNameDeclaration extends AbstractNameDeclaration {
 
-    public MethodNameDeclaration(ASTMethodDeclarator node) {
+    public MethodNameDeclaration(ASTMethodDeclaration node) {
```

Comment:
Hm... that's an API change. But I would consider the constructor as internal API, that should not be called from rule implementors. We probably need to internalize that on master?

Reply:
I think the whole package should be deprecated for removal tbh. #1566 is a superior alternative.

Reply:
ok, let's postpone this deprecation until we have a stable API for the new symboltable.